pvlib/pvsystem.py

```diff
@@ -223,12 +251,49 @@ def __init__(self,
             )
 
     def __repr__(self):
-        attrs = ['name', 'surface_tilt', 'surface_azimuth', 'module',
-                 'inverter', 'albedo', 'racking_model', 'module_type',
-                 'temperature_model_parameters']
-        return ('PVSystem:\n  ' + '\n  '.join(
-            f'{attr}: {getattr(self, attr)}' for attr in attrs))
+        repr = f'PVSystem:\n  name: {self.name}\n  '
+        for array in self._arrays:
+            repr += '\n  '.join(array.__repr__().split('\n'))
+            repr += '\n  '
+        repr += f'inverter: {self.inverter}'
+        return repr
+
+    def _validate_per_array(self, values, system_wide=False):
```

Comment:
Just a comment about the name: this function does more than validating that the lengths are the same. It also creates a tuple if `values` is not a tuple, and broadcasts if `system_wide` is True.  To me `validate` implies returning True/False.  I don't have an alternate name to suggest.

Reply:
I agree with this, but I cannot think of a better name either.

Reply:
The best I've got would be just `_per_array()`.